source/simplex/fe_lib.cc

```diff
@@ -559,6 +585,158 @@ namespace Simplex
 
 
 
+  template <int dim, int spacedim>
+  FE_PyramidP<dim, spacedim>::FE_PyramidP(const unsigned int degree)
+    : dealii::FE_Poly<dim, spacedim>(
+        Simplex::ScalarPyramidPolynomial<dim>(degree),
+        FiniteElementData<dim>(get_dpo_vector_fe_pyramid(degree),
+                               ReferenceCell::Type::Pyramid,
+                               1,
+                               degree,
+                               FiniteElementData<dim>::L2),
+        std::vector<bool>(FiniteElementData<dim>(get_dpo_vector_fe_pyramid(
+                                                   degree),
+                                                 ReferenceCell::Type::Pyramid,
+                                                 1,
+                                                 degree)
+                            .dofs_per_cell,
+                          true),
+        std::vector<ComponentMask>(
+          FiniteElementData<dim>(get_dpo_vector_fe_pyramid(degree),
+                                 ReferenceCell::Type::Pyramid,
+                                 1,
+                                 degree)
+            .dofs_per_cell,
+          std::vector<bool>(1, true)))
+  {
+    AssertDimension(dim, 3);
+
+
+    if (degree == 1)
+      {
+        this->unit_support_points.emplace_back(-1.0, -1.0, 0.0);
+        this->unit_support_points.emplace_back(+1.0, -1.0, 0.0);
+        this->unit_support_points.emplace_back(-1.0, +1.0, 0.0);
+        this->unit_support_points.emplace_back(+1.0, +1.0, 0.0);
+        this->unit_support_points.emplace_back(+0.5, +0.5, 1.0);
```

Comment:
This point seems wrong - shouldn't the `z = 1` point be at the origin if we use `[-1, 1]`?

Reply:
Indeed.